eanalyze_staging.sql

```diff
@@ -0,0 +1,1483 @@
+  `follows_up_from` int DEFAULT NULL,
+  `is_official` int DEFAULT '1',
+  PRIMARY KEY (`shipment_id`),
+  KEY `scheme_type` (`scheme_type`),
+  KEY `distribution_id` (`distribution_id`),
+  KEY `shipment_ibfk_3_idx` (`follows_up_from`),
+  KEY `shipment_id` (`shipment_id`),
+  CONSTRAINT `shipment_ibfk_1` FOREIGN KEY (`scheme_type`) REFERENCES `scheme_list` (`scheme_id`),
+  CONSTRAINT `shipment_ibfk_3` FOREIGN KEY (`follows_up_from`) REFERENCES `shipment` (`shipment_id`) ON DELETE SET NULL,
+  CONSTRAINT `shipment_ibfk_4` FOREIGN KEY (`distribution_id`) REFERENCES `distributions` (`distribution_id`) ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `shipment_participant_map`
+--
+
+DROP TABLE IF EXISTS `shipment_participant_map`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `shipment_participant_map` (
+  `map_id` int NOT NULL AUTO_INCREMENT,
+  `shipment_id` int NOT NULL,
+  `participant_id` int NOT NULL,
+  `attributes` mediumtext,
+  `evaluation_status` varchar(10) DEFAULT NULL COMMENT 'Shipment Status					\nUse this to flag - 					\nABCDEFG					',
+  `shipment_score` decimal(5,2) DEFAULT NULL,
+  `documentation_score` decimal(5,2) DEFAULT '0.00',
+  `shipment_test_date` date DEFAULT NULL,
+  `is_pt_test_not_performed` varchar(45) DEFAULT NULL,
+  `pt_test_not_performed_comments` text,
+  `pt_support_comments` text,
+  `shipment_receipt_date` date DEFAULT NULL,
+  `shipment_test_report_date` datetime DEFAULT NULL,
+  `participant_supervisor` varchar(255) DEFAULT NULL,
+  `supervisor_approval` varchar(45) DEFAULT NULL,
+  `review_date` date DEFAULT NULL,
+  `final_result` int DEFAULT '0',
+  `failure_reason` text,
+  `evaluation_comment` int DEFAULT '0',
+  `optional_eval_comment` text,
+  `is_followup` varchar(255) DEFAULT 'no',
+  `is_excluded` varchar(255) NOT NULL DEFAULT 'no',
+  `user_comment` text,
+  `custom_field_1` text,
+  `custom_field_2` text,
+  `created_on_admin` datetime DEFAULT NULL,
+  `updated_on_admin` datetime DEFAULT NULL,
+  `updated_by_admin` varchar(45) DEFAULT NULL,
+  `updated_on_user` datetime DEFAULT NULL,
+  `updated_by_user` varchar(45) DEFAULT NULL,
+  `created_by_admin` varchar(45) DEFAULT NULL,
+  `created_on_user` datetime DEFAULT NULL,
+  `report_generated` varchar(100) DEFAULT NULL,
+  `last_new_shipment_mailed_on` datetime DEFAULT NULL,
+  `new_shipment_mail_count` int NOT NULL DEFAULT '0',
+  `last_not_participated_mailed_on` datetime DEFAULT NULL,
+  `last_not_participated_mail_count` int NOT NULL DEFAULT '0',
+  `qc_done` varchar(45) NOT NULL DEFAULT 'no',
+  `qc_date` date DEFAULT NULL,
+  `qc_done_by` varchar(255) DEFAULT NULL,
+  `qc_created_on` datetime DEFAULT NULL,
+  `mode_id` int DEFAULT NULL,
+  `not_tested_reason` int DEFAULT NULL,
+  `date_submitted` datetime DEFAULT NULL,
+  PRIMARY KEY (`map_id`),

```

Comment:
The existing database has a unique constraint which does not exist in this script. Please add it?
```
UNIQUE KEY `shipment_id_2` (`shipment_id`,`participant_id`),
```

Reply:
okay let add the key